src/brod_group_subscriber.erl

```diff
@@ -35,7 +35,9 @@
 %%%=============================================================================
 
 -module(brod_group_subscriber).
+
 -behaviour(gen_server).
+-behaviour(brod_group_controller).
```

Comment:
Why 'must'?
a group subscriber is a message subscriber, a group controller.
A group subscriber should implement certain callbacks for group controller to interact with.

This doesn't necessary to be a behaviour, having them explicitly declared is easier for compiler/dialyzer checks.

If this only about the name being misleading, how about rename controller to coordinator ?